app/models/bank_transaction.rb

```diff
@@ -33,4 +33,56 @@ class BankTransaction < ActiveRecord::Base
   def image_url
     'data:image/png;base64,' + Base64.encode64(self.image)
   end
+
+  def assign_to_invoice
+    return false unless supplier
+
+    content = text
+    content += "\n" + reference if reference.present?
+    invoices = supplier.invoices.unpaid.select {|i| content.include? i.number}
```

Comment:
Is `include?` exact enough? What it there are invoices for number `12` and `123`? I might introduce a certain format for this.

Reply:
i agree that it's not 100% safe, but the usability must be high at this place, from my experience. if multiple users are responsible for doing the bank transfers it's hard that everybody does it with the exact same format.
since we have an additional check with the amount of the invoices it should be enough

Reply:
if there is any chance an automated match could be wrong, you should probably not do it.  it can be fine as something in the ui to suggest relevant matches, but if there is any chance of error. 

as an aside, one 'hack' we do in our co-op is asking people to always put some extra cents on the end to make their transaction unique.  with e-transfers, etc, it isn't always clear where the money comes from when looking at the bank statement.